### ISSUE TITLE
[Bug]: When the parameter of function cos is  constant null, the function matching selection will report an error within the plan2 

### ISSUE THREAD

**Issue by qingxinhome** [matrixorigin/matrixone#2790]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

mysql> select cos(null) from d_table1;
ERROR 1105 (HY000): unknown error:[42601]too much function matches
cos[TINYINT UNSIGNED]
cos[SMALLINT UNSIGNED]
cos[INT UNSIGNED]
cos[BIGINT UNSIGNED]
cos[TINYINT]
cos[SMALLINT]
cos[INT]
cos[BIGINT]
cos[FLOAT]
cos[DOUBLE]

### Expected Behavior

mysql> select cos(null) from d_table1;
+-----------+
| cos(null) |
+-----------+
|      NULL |
|      NULL |
|      NULL |
+-----------+

**Comment by qingxinhome** [comment]
The same problem exists:
'atan'
'cos' 
'cot' 
'sin' 
'sinh'
'tan' 
'ln'  
'month'

**Comment by m-schen** [comment]
I have fixed this issue, plz verify it after #2953 was merged.

**Comment by qingxinhome** [comment]
> I have fixed this issue, plz verify it after #2953 was merged.

Ok，nice

**Comment by m-schen** [comment]
> > I have fixed this issue, plz verify it after #2953 was merged.
> 
> Ok，nice

thanks, #2953 was merged into branch Main now.
